core/src/Serial/Kokkos_Serial.hpp

```diff
@@ -61,6 +64,10 @@ class SerialInternal {
   static SerialInternal& singleton();
 
   std::mutex m_thread_team_data_mutex;
+  std::mutex m_instance_mutex;
```

Comment:
(Drop `m_thread_team_data_mutex`)

Reply:
We need two locks since resize the scratch data might call fence which in turn locks the instance mutex so we are deadlocking if locking the instance mutex before resizing.